tests/end_to_end/workflow/subset_flow.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import random
+from openfl.experimental.workflow.interface.fl_spec import FLSpec
+from openfl.experimental.workflow.placement.placement import aggregator, collaborator
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class TestFlowSubsetCollaborators(FLSpec):
+    """
+    Testflow to validate working of Subset Collaborators in Federated Flow.
+    """
+
+    def __init__(self, random_ints=[], **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.random_ints = random_ints
+
+    @aggregator
+    def start(self):
+        """
+        Starting the flow with random subset of collaborators
+        """
+        log.info("Testing FederatedFlow - Starting Test for validating Subset of collaborators")
+        self.collaborators = self.runtime.collaborators
+
+        # select subset of collaborators
+        self.subset_collaborators = self.collaborators[: random.choice(self.random_ints)]
+
+        log.info(f"... Executing flow for {len(self.subset_collaborators)} collaborators out of Total: {len(self.collaborators)}")
+
+        self.next(self.test_valid_collaborators, foreach="subset_collaborators")
+
+    @collaborator
+    def test_valid_collaborators(self):
+        """
+        set the collaborator name
+        """
+        log.info(f"Print collaborators {self.name}")
+        self.collaborator_ran = self.name
+        self.next(self.join)
+
+    @aggregator
+    def join(self, inputs):
+        """
+        List of collaborators ran successfully
+        """
+        log.info("inside join")
```

Comment:
Please remove this print stmt or move to log.debug.

Reply:
These statements are used in MetaFlow I think. This is how it checks where all it went.

Reply:
No, there's no such statement in Metaflow - https://github.com/search?q=repo%3ANetflix%2Fmetaflow+%22inside+join%22&type=code

Even if we want to log here, let's print the list of collaborators which ran here

log.info(f"Inside join function with {self.collaborators_ran}")